src/aave-v2/lens/RatesLens.sol

```diff
@@ -359,9 +417,41 @@ abstract contract RatesLens is UsersLens {
         );
     }
 
-    /// INTERNAL ///
+    /// @dev Computes and returns the underlying pool rates for a specific market.
+    /// @param _underlying The underlying pool market address.
+    /// @param _supplied The amount hypothetically supplied (in underlying).
+    /// @param _borrowed The amount hypothetically borrowed (in underlying).
+    /// @param _repaid The amount hypothetically repaid (in underlying).
+    /// @param _withdrawn The amount hypothetically withdrawn (in underlying).
+    /// @return poolSupplyRate The market's pool supply rate per year (in ray).
+    /// @return poolBorrowRate The market's pool borrow rate per year (in ray).
+    function _getPoolRatesPerYear(
+        address _underlying,
+        uint256 _supplied,
+        uint256 _borrowed,
+        uint256 _withdrawn,
+        uint256 _repaid
+    ) internal view returns (uint256 poolSupplyRate, uint256 poolBorrowRate) {
+        DataTypes.ReserveData memory reserve = pool.getReserveData(_underlying);
+        (, , , , uint256 reserveFactor) = reserve.configuration.getParamsMemory();
+
+        (poolSupplyRate, , poolBorrowRate) = IReserveInterestRateStrategy(
+            reserve.interestRateStrategyAddress
+        ).calculateInterestRates(
+            _underlying,
+            ERC20(_underlying).balanceOf(reserve.aTokenAddress) +
+                _supplied +
+                _repaid -
+                _borrowed -
+                _withdrawn,
+            ERC20(reserve.stableDebtTokenAddress).totalSupply(),
```

Comment:
Isn't it supposed to be the `totalStableDebt` here ? I guess that it is useless here, and that would be why it is not breaking tests. I would rather send 0.

Reply:
It is supposed to `totalStableDebt` here and indeed it is, because `totalStableDebt = ERC20(reserve.stableDebtTokenAddress).totalSupply()`: https://github.com/aave/protocol-v2/blob/ce53c4a8c8620125063168620eba0a8a92854eb8/contracts/misc/AaveProtocolDataProvider.sol#L95-L126